truss/main.go

```diff
@@ -6,133 +6,209 @@ import (
 	"io"
 	"io/ioutil"
 	"os"
-	"path"
 	"path/filepath"
 	"strings"
 
 	"github.com/pkg/errors"
 
-	"github.com/TuneLab/go-truss/truss/protostage"
+	"github.com/TuneLab/go-truss/truss/protostuff"
 	"github.com/TuneLab/go-truss/truss/truss"
 
 	"github.com/TuneLab/go-truss/deftree"
 	"github.com/TuneLab/go-truss/gendoc"
 	"github.com/TuneLab/go-truss/gengokit"
 )
 
-func main() {
+var (
+	pbPathFlag = flag.String("pbout", "", "The go package path where the protoc-gen-go .pb.go structs will be written.")
+)
+
+func init() {
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "Usage: %s [OPTION]... [*.proto]...\n", os.Args[0])
+		flag.PrintDefaults()
+	}
+
 	flag.Parse()
 
 	if len(flag.Args()) == 0 {
-		exitIfError(errors.New("no arguments passed"))
+		fmt.Fprintf(os.Stderr, "%s: missing .proto file(s)\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "Try '%s --help' for more information.\n", os.Args[0])
+		os.Exit(1)
 	}
+}
+
+func main() {
+	cfg, err := parseInput()
+	exitIfError(errors.Wrap(err, "could not parse input"))
+
+	dt, err := parseServiceDefinition(cfg.DefPaths)
+	exitIfError(errors.Wrap(err, "unable to parse input definition proto files"))
+
+	err = updateConfigWithService(cfg, dt)
+	exitIfError(errors.Wrap(err, "unable to read system based on service definition"))
+
+	genFiles, err := generateCode(cfg, dt)
+	exitIfError(errors.Wrap(err, "unable to generate service"))
 
+	for _, f := range genFiles {
+		err := writeGenFile(f, cfg.ServicePath)
+		if err != nil {
+			exitIfError(errors.Wrap(err, "unable to write output"))
+		}
+	}
+}
+
+// parseInput constructs a *truss.Config with all values needed to parse
+// service definition files.
+func parseInput() (*truss.Config, error) {
+	var cfg truss.Config
+
+	// GOPATH
+	goPaths := filepath.SplitList(os.Getenv("GOPATH"))
+	if goPaths == nil {
+		return nil, errors.New("$GOPATH not set")
+	}
+	cfg.GOPATH = goPaths[0]
+
+	// DefPaths
+	var err error
 	rawDefinitionPaths := flag.Args()
+	cfg.DefPaths, err = cleanProtofilePath(rawDefinitionPaths)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to parse input arguments")
+	}
 
-	protoDir, definitionFiles, err := cleanProtofilePath(rawDefinitionPaths)
+	// PBGoPath
+	if *pbPathFlag != "" {
```

Comment:
I personally like the inversion, flattens things out even if the return is repeated. 

I chose all caps for `GOPATH` because that is what it is called in the go build context struct: https://github.com/golang/go/blob/master/src/go/build/build.go#L34